strawberry_django/fields/filter_types.py

```diff
@@ -74,10 +74,41 @@ class FilterLookup(BaseFilterLookup[T]):
     i_ends_with: T | None = filter_field(
         description=f"Case-insensitive ends-with. {_SKIP_MSG}"
     )
-    regex: T | None = filter_field(
+    regex: str | None = filter_field(
         description=f"Case-sensitive regular expression match. {_SKIP_MSG}"
     )
-    i_regex: T | None = filter_field(
+    i_regex: str | None = filter_field(
+        description=f"Case-insensitive regular expression match. {_SKIP_MSG}"
+    )
+
+
+@strawberry.input
+class UUIDFilterLookup(BaseFilterLookup[T]):
```

Comment:
So I made another filter. For now it's used only for UUID filtering, but actually could be used for other types. 

I have tried to reuse `FilterLookup[str]` but got issues with that, so I think for now it's better to keep it simple. 
1. Used that only for filtering UUID and named it as the place where it used. If there will be other cases where it can be used, then it can be renamed accordingly. (Well, I haven’t yet been able to think of anything else besides filtering as text for UUID fields)
2. To have some code duplication, but have correct annotations. (without monkey patching, overrides at related packages, extra branches in the code, dark magic)

But @bellini666 if you know the say how to make it more eloquent feel free to share and I will be happy to refactor.